src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ReplayDataPlaybackController.cs

```diff
@@ -133,6 +134,12 @@ public void Stop()
 
         private double CurrentTimePoint => Time.unscaledTime - _lastStartTime + _priorKeyFrameTime ?? 0.0;
 
+        // used to avoid spam building/showing the keyframe text until at least a few frames have gone past
+        private int _checkOfKeyFrameCount = 0;
+
+        // allow up to 360 frames before saying it failed, at 120 fps this is 3 seconds, at 100fps this is ~3.6 seconds , at 60fps this is 6 seconds , at 50 fps this is ~7.2 seconds, at 30 fps this is 12 seconds
+        private const int KeyFrameChecksBeforePrompting = 360;
```

Comment:
(non-blocking) Should this be a public variable that uses can edit? Might be nice to control the timeout (for instance, maybe I know that my network latency results in a highly-varied wait time)

Reply:
in short no, this is a nicety to avoid some screen flicker but does not affect whether this system functions or not

Reply:
Ahhh I see, thanks for the clarification.